hooks/useEmailManagement.ts

```diff
@@ -0,0 +1,289 @@
+  const updateUserEmail = async (email: string, verificationToken: string) => {
+    if (Platform.OS === 'web') {
+      const { Turnkey } = await import('@turnkey/sdk-browser');
+      const turnkey = new Turnkey({
+        apiBaseUrl: EXPO_PUBLIC_TURNKEY_API_BASE_URL,
+        defaultOrganizationId: EXPO_PUBLIC_TURNKEY_ORGANIZATION_ID,
+        rpId: getRuntimeRpId(),
+      });
+
+      const passkeyClient = turnkey.passkeyClient();
+      const indexedDbClient = await turnkey.indexedDbClient();
+      await indexedDbClient?.init();
+      await indexedDbClient!.resetKeyPair();
+      const pubKey = await indexedDbClient!.getPublicKey();
+      await passkeyClient?.loginWithPasskey({
+        sessionType: SessionType.READ_WRITE,
+        publicKey: pubKey!,
+      });
+      await passkeyClient?.updateUserEmail({
+        userId: user?.userId as string,
+        userEmail: email,
+        verificationToken,
+      });
+    } else {
+      const stamper = new PasskeyStamper({
+        rpId: getRuntimeRpId(),
+      });
+
+      const turnkeyClient = new TurnkeyClient(
+        { baseUrl: EXPO_PUBLIC_TURNKEY_API_BASE_URL },
+        stamper,
+      );
+
+      await turnkeyClient.updateUserEmail({
+        type: 'ACTIVITY_TYPE_UPDATE_USER_EMAIL',
+        timestampMs: Date.now().toString(),
+        organizationId: user?.suborgId as string,
+        parameters: {
+          userId: user?.userId as string,
+          userEmail: email,
+          verificationToken,
+        },
+      });
+    }
+  };
+
+  const handleSendOtp = async (data: EmailFormData) => {
+    setIsLoading(true);
+    setRateLimitError(null);
+    try {
+      const response = await initGenericOtp(data.email, 6, false);
+      setOtpId(response.otpId);
+      setEmailValue(data.email);
+      setStep('otp');
+      if (Platform.OS !== 'web') {
+        Alert.alert('OTP Sent', 'A verification code has been sent to your email address.', [
+          { text: 'OK' },
+        ]);
+      }
+    } catch (error: any) {
+      const errorMessage = error?.message || error?.toString() || '';
+      const isRateLimitError = errorMessage.includes('Max number of OTPs have been initiated') ||
+        errorMessage.includes('please wait and try again') ||
+        errorMessage.includes('Turnkey error 3');
+
+      if (isRateLimitError) {
+        setRateLimitError('Too many verification codes requested. Please wait a few minutes before trying again.');
+      } else {
+        if (Platform.OS !== 'web') {
+          Alert.alert('Error', 'Failed to send verification code. Please try again.', [{ text: 'OK' }]);
+        }
+      }
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleVerifyOtp = async (data: OtpFormData) => {
+    setIsLoading(true);
+    try {
+      const verifyResponse = await verifyGenericOtp(otpId, data.otpCode);
+
+      await updateUserEmail(emailValue, verifyResponse.verificationToken);
+      await withRefreshToken(() => updateEmail(emailValue));
```

Comment:
@LiorAgnin update email in the backend when OTP is verified.

Currently OTP verification can be bypassed, by directly calling /update-email endpoint without verificationToken.

Reply:
@MusabShakeel576  Good idea! Thanks for spotting this.